### ISSUE TITLE
Is Pyarmor not supported for Isilon OneFS OS ? 

### ISSUE THREAD

**Issue by Soujanya459** [dashingsoft/pyarmor#508]
Getting below error when I run the binary file(obfuscation done using Pyarmor) in 'Isilon OneFS' operating system. 
'Platform isilon onefs not supported'

Please provide fix for the issue.. I have explained issue over the mail too..

Thank You
Soujanya

**Comment by jondy** [comment]
No, pyarmor doesn't support this platform. I don't understand this platform, but if it's based on linux source, maybe other platform would work.
More information refer to
https://pyarmor.readthedocs.io/en/latest/platforms.html

**Comment by jondy** [comment]
After I got to know a little about Isilon OneFS, it seems it's based on FreeBSD, so does this command work in this platform
```
pyarmor obfuscate --platform freebsd.x86_64 foo.py
python dist/foo.py
```

If it works, it's easy to support for this platform.

**Comment by Soujanya459** [comment]
/root/.pyarmor/platforms/index.json --- > doesn't have platform information related to OneFS 

ese-latest-1# pyarmor obfuscate --platform freebsd.x86_64 foo.py ------ in OneFS 
INFO     Search library for platform: isilon onefs.x86_64
WARNING  The trial version could not download the latest core libraries, tag r41.15a is always used
WARNING  The core library excepted version is r42.16, but got r41.15a from platform list file /root/.pyarmor/platforms/index.json
ERROR    Unsupport platform isilon onefs.x86_64
ERROR    No available library for this platform


I got the same error in OneFS during License registration too...

------------->>>>>>>>>>>>>>
root@FreeBSD113:~ # pyarmor obfuscate --platform freebsd.x86_64 fooTest.py  ----------> In FreeBSD113
INFO     PyArmor Version 6.6.0
INFO     Target platforms: ['freebsd.x86_64']
INFO     Update target platforms to: ['freebsd.x86_64.0']
-----
INFO     Start obfuscating the scripts...
INFO            OneFS_pytransform.py -> dist/OneFS_pytransform.py
INFO            fooTest.py -> dist/fooTest.py
INFO     Insert bootstrap code to entry script dist/fooTest.py
INFO     Obfuscate 2 scripts OK.

**Comment by jondy** [comment]
Do these commands work in OneFS?
```
PYARMOR_PLATFORM=freebsd.x86_64 pyarmor obfuscate foo.py
python dist/foo.py 
```

**Comment by Soujanya459** [comment]
ese-latest-1# PYARMOR_PLATFORM=freebsd.x86_64 pyarmor obfuscate foo.py
INFO     PYARMOR_PLATFORM is freebsd.x86_64
INFO     Search library for platform: freebsd.x86_64
WARNING  The trial version could not download the latest core libraries, tag r41.15a is always used
WARNING  The core library excepted version is r42.16, but got r41.15a from platform list file /root/.pyarmor/platforms/index.json
INFO     Found available libraries: ['freebsd.x86_64.0']
INFO     Target path for freebsd.x86_64.0: /root/.pyarmor/platforms/freebsd/x86_64/0
INFO     Downloading library file for freebsd.x86_64.0 ...
WARNING  The trial version could not download the latest platform library
INFO     Getting remote file: https://github.com/dashingsoft/pyarmor-core/raw/r41.15a/platforms/freebsd/_pytransform.so
INFO     Writing target file: /root/.pyarmor/platforms/freebsd/x86_64/0/_pytransform.so
INFO     Download dynamic library freebsd.x86_64.0 OK
ERROR    Platform isilon onefs not supported

**Comment by jondy** [comment]
What's the output of `ldd /root/.pyarmor/platforms/freebsd/x86_64/0/_pytransform.so`?

**Comment by Soujanya459** [comment]
ese-latest-1# ldd /root/.pyarmor/platforms/freebsd/x86_64/0/_pytransform.so  ----> OneFS
/root/.pyarmor/platforms/freebsd/x86_64/0/_pytransform.so:
        libc.so.7 => /lib/libc.so.7 (0x800248000)
ese-latest-1#


root@FreeBSD113:~ # ldd /root/.pyarmor/platforms/freebsd/x86_64/0/_pytransform.so --> FreeBSD 
/root/.pyarmor/platforms/freebsd/x86_64/0/_pytransform.so:
        libc.so.7 => /lib/libc.so.7 (0x800825000)
root@FreeBSD113:~ #

**Comment by jondy** [comment]
well, it's easy to support this platform. A quick workaround is apply this patch 209642928c40bdbe7dfb51e3d7a5d5ce52abff6b

It will be released in next version (v6.7.1)

**Comment by Soujanya459** [comment]
Hi jondy,

Not only this fix  ('freebsd', ('freebsd*', 'openbsd*', 'isilon onefs')),
but also fix in the line #305 as below helps .. 

elif plat == 'freebsd' or 'onefs' in plat:

**Comment by jondy** [comment]
It's not required, because the platform name will be "freebsd" when pyarmor runs in this platform

**Comment by Soujanya459** [comment]
('freebsd', ('freebsd*', 'openbsd*', 'isilon onefs')), -- After this patch I got the error 'Platform isilon onefs not supported'

After editing the line #305 -- it worked fine because platform.system() returns value as 'Isilon OneFS'

**Comment by jondy** [comment]
Sorry, you're right. But I prefer to this patch 0c325245afa89e42f37be29195c488bd540bd27f

**Comment by Soujanya459** [comment]
Hi Jondy

This 0c32524 patch worked fine..

we have immediate release of our product.

1. can you please provide me which version of pyarmor will have this fix and ETA for the same?

2. If release is getting delayed shall I go ahead build our product with patch and deliver to customer,
hope it won't cause any license problem ? License is in the name of 'Pradeep Kuppam' Dell Technologies.

Thank You,
Soujanya

**Comment by jondy** [comment]
Here it's the pre-release of this new version v6.7.1, it may be released in a few days
https://pyarmor.readthedocs.io/en/latest/change-logs.html#dev

**Comment by jondy** [comment]
Fix it in v6.7.1, it has released.

**Comment by Soujanya459** [comment]
ThankYou Jondy.
